jdk/src/solaris/back/linker_md.c

```diff
@@ -54,9 +60,10 @@
 #define LIB_SUFFIX "so"
 #endif
 
-static void dll_build_name(char* buffer, size_t buflen,
+static void dll_build_name(char* buffer, int buflen,
```

Comment:
This is inappropriate: `size_t` is the right type to use here.

Reply:
There was me trying to avoid a similar error in dll_build_name. I presumed it'd "warn" me when it tried compiling the narrowing conversion later on (in the dll_build_name call).

Change in. Running build.